integration-tests/src/test/java/com/example/IntegrationTest.java

```diff
@@ -835,6 +835,27 @@ public void multibindingSetElements() {
     assertThat(component.values()).containsExactly("one", "two");
   }
 
+  @Test
+  @IgnoreCodegen
+  public void multibindingSetConflictingExplicitAndImplicit() {
+    try {
+      backend.create(MultibindingSetConflict.class);
+      fail();
+    } catch (IllegalStateException e) {
+      // Note: using only one of each IntoSet and ElementsIntoSet to make sure the test is stable,
+      // because Class.getDeclaredMethods() doesn't guarantee ordering.
+      assertThat(e)
+          .hasMessageThat()
+          .isEqualTo(
+              "java.util.Set<java.lang.String> has incompatible bindings or declarations:\n"
+                  + "  Set bindings and declarations:\n"
+                  + "    [single  ] @Provides[com.example.MultibindingSetConflict$Module1.one(…)]\n"
+                  + "    [multiple] @Provides[com.example.MultibindingSetConflict$Module1.two(…)]\n"
```

Comment:
What about "instance" and "elements"? We can re-use terminology like "elements" which is in the API of `Scope`, I just don't want explicitly mechanisms of Dagger like `@ElementsIntoSet` (unless we sneak them into the `Binding`'s `toString()`, then we can have as much Dagger-specific things as we like).